android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt

```diff
@@ -420,31 +420,6 @@ constructor(
     completed * 100 / if (total > 0) total else 1
 
   suspend fun schedulePeriodicJobs(context: Context) {
-    if (!BuildConfig.SKIP_AUTHENTICATION) {
-      if (context.isDeviceOnline()) {
-        // Do not schedule sync until location selected when strategy is RelatedEntityLocation
-        // Use applicationConfiguration.usePractitionerAssignedLocationOnSync to identify
-        // if we need to trigger sync based on assigned locations or not
-        when {
-          applicationConfiguration.syncStrategy.contains(SyncStrategy.RelatedEntityLocation) -> {
-            if (
-              applicationConfiguration.usePractitionerAssignedLocationOnSync ||
-                context
-                  .retrieveRelatedEntitySyncLocationState(MultiSelectViewAction.SYNC_DATA)
-                  .isNotEmpty()
-            ) {
-              syncBroadcaster.schedulePeriodicSync(applicationConfiguration.syncInterval)
-            }
-          }
-          else -> syncBroadcaster.schedulePeriodicSync(applicationConfiguration.syncInterval)
-        }
-      } else {
-        withContext(dispatcherProvider.main()) {
-          context.showToast(context.getString(R.string.sync_failed), Toast.LENGTH_LONG)
-        }
-      }
-    }
-
```

Comment:
could you check the relevancy of this code to the periodic worker code?
and see if there's a reason on why the order of the code matters

Reply:
By process of elimination, the cause has been narrowed down to this line
`syncBroadcaster.schedulePeriodicSync(applicationConfiguration.syncInterval)` .
The `why` is yet to be determined.